src/test/java/TestUtils.java

```diff
@@ -95,17 +95,33 @@ public static void setSearchServletRequestParameters(HttpServletRequest request,
     when(request.getParameter("max")).thenReturn(maxPrice);
   }
 
-  /** Parses a string containing store analytics into a hashmap representation. */
+  /**
+   * Parses a string containing store analytics into a hashmap representation.
+   * str is in the form: {"storeAnalytics":{...}, "categoryAnalytics": {...}}
+   */
   public static HashMap<String, Double> parseStoreAnalytics(String str) throws IOException {
-    String stores = str.substring(str.indexOf("{"), str.indexOf("}") + 1);
-    stores = stores.substring(stores.lastIndexOf("{"), stores.indexOf("}") + 1);
-    return new ObjectMapper().readValue(stores, HashMap.class);
+    String key = "storeAnalytics";
+
+    // Set indexes to extract the first {...} from str.
+    int startIndex = str.indexOf(key) + key.length() + 2;
+    int endIndex = str.indexOf("}") + 1;
+
+    String json = str.substring(startIndex, endIndex);
+    return new ObjectMapper().readValue(json, HashMap.class);
```

Comment:
Do you think this would be a better way to parse the string?
https://stackoverflow.com/questions/2591098/how-to-parse-json-in-java